deployment/src/tests/test_lambda.py

```diff
@@ -0,0 +1,292 @@
+import json
+import os
+import random
+
+import pulumi.runtime
+import pytest
+
+from strongmind_deployment.operations import get_code_owner_team_name
+from tests.mocks import get_pulumi_mocks
+from strongmind_deployment.lambda_component import LambdaComponent, LambdaArgs, LambdaEnvVariables
+from tests.shared import assert_outputs_equal, assert_output_equals
+
+
+def describe_lambda_args():
+    @pytest.fixture
+    def handler(faker):
+        return faker.word()
+
+    @pytest.fixture
+    def runtime(faker):
+        valid_runtimes = [
+            "python3.9",
+            "python3.10",
+            "python3.11",
+        ]
+        return random.choice(valid_runtimes)
+
+    @pytest.fixture
+    def timeout(faker):
+        return faker.random_int()
+
+    @pytest.fixture
+    def memory_size(faker):
+        return faker.random_int()
+
+    @pytest.fixture
+    def layers(faker):
+        return [faker.word(), faker.word()]
+
+    @pytest.fixture
+    def sut(handler, runtime, timeout, memory_size, layers):
+        return LambdaArgs(
+            handler=handler,
+            runtime=runtime,
+            timeout=timeout,
+            memory_size=memory_size,
+            layers=layers,
+        )
+
+    @pulumi.runtime.test
+    def it_exists(sut):
+        assert sut
+
+    @pulumi.runtime.test
+    def it_has_a_handler(sut, handler):
+        assert sut.handler == handler
+
+    @pulumi.runtime.test
+    def it_has_a_runtime(sut, runtime):
+        assert sut.runtime == runtime
+
+    @pulumi.runtime.test
+    def it_has_a_timeout(sut, timeout):
+        assert sut.timeout == timeout
+
+    @pulumi.runtime.test
+    def it_has_a_memory_size(sut, memory_size):
+        assert sut.memory_size == memory_size
+
+    @pulumi.runtime.test
+    def it_has_layers(sut, layers):
+        assert sut.layers == layers
+
+
+def describe_lambda_environment():
+    @pytest.fixture
+    def variables(faker):
+        return {}
+
+    @pytest.fixture
+    def sut(variables):
+        return LambdaEnvVariables(variables)
+
+    @pulumi.runtime.test
+    def it_exists(sut):
+        assert sut
+
+    @pulumi.runtime.test
+    def it_has_a_dict_of_variables(sut):
+        assert isinstance(sut.variables, dict)
```

Comment:
It's more valuable here to check the value - put some kind of mock in the variables fixture, and then compare the system under test to the variables you sent it.